src/unix/lwt_engine.ml

```diff
@@ -197,6 +197,55 @@ end
 
 class libev_deprecated = libev ()
 
+class libuv () = object
+  inherit abstract
+
+  val loop = Luv.Loop.default ()
+
+  method loop = loop
+
+  method private cleanup = Luv.Loop.stop loop
+
+  method iter block =
+    try
+      match (block) with
+      | true -> Luv.Loop.run ~loop ~mode:`ONCE () |> ignore
+      | false -> Luv.Loop.run ~loop ~mode:`NOWAIT () |> ignore
+    with exn ->
+      Luv.Loop.stop loop;
+      raise exn
+
+  method private register_readable fd f =
+    let p = Luv.Poll.init ~loop (Obj.magic fd) in
```

Comment:
This version still has the "naked" `Obj.magic`; did you mean to include factoring it out in this commit?

Reply:
Fixed in a new commit